packages/netlify-cms-core/src/components/Collection/Collection.js

```diff
@@ -74,10 +94,36 @@ class Collection extends React.Component {
 
 function mapStateToProps(state, ownProps) {
   const { collections } = state;
-  const { isSearchResults, match } = ownProps;
+  const { isSearchResults, match, t } = ownProps;
   const { name, searchTerm } = match.params;
   const collection = name ? collections.get(name) : collections.first();
-  return { collection, collections, collectionName: name, isSearchResults, searchTerm };
+  const sort = selectEntriesSort(state.entries, collection.get('name'));
+  const sortableFields = selectSortableFields(collection, t);
+
+  return {
+    collection,
+    collections,
+    collectionName: name,
+    isSearchResults,
+    searchTerm,
+    sort,
+    sortableFields,
+  };
 }
 
-export default connect(mapStateToProps)(Collection);
+const mapDispatchToProps = {
+  sortByField,
+};
+
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
+  return {
+    ...stateProps,
+    ...ownProps,
+    onSortClick: (key, direction) =>
+      dispatchProps.sortByField(stateProps.collection, key, direction),
+  };
+};
+
+const ConnectedCollection = connect(mapStateToProps, mapDispatchToProps, mergeProps)(Collection);
```

Comment:
Moved the connected sorting stuff here in favor of a dumb sort component.

Reply:
I liked the ability to sort with one click though (and have the current sort always displayed).

Reply:
I think I read this https://uxmovement.com/buttons/why-segmented-buttons-are-better-filters-than-dropdowns/ before deciding on the UI

Reply:
This comment was more about removing the redux `connect` method from the sort component, but yeah I'm in agreement with the approach in the article for sorting in general. For the CMS, though, sorting will likely be a set-it-and-forget-it tool, in which case it makes sense for it to be simple and out of the way. The main goal for this first take should be simplicity, but we can definitely iterate moving forward.